pkg/resources/materialized_view.go

```diff
@@ -0,0 +1,330 @@
+
+	db := meta.(*sql.DB)
+	if d.HasChange("name") {
+		name := d.Get("name")
+
+		q := builder.Rename(name.(string))
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error renaming view %v", d.Id())
+		}
+
+		d.SetId(fmt.Sprintf("%v|%v|%v", dbName, schema, name.(string)))
+	}
+
+	if d.HasChange("comment") {
+		comment := d.Get("comment")
+
+		if c := comment.(string); c == "" {
+			q := builder.RemoveComment()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error unsetting comment for view %v", d.Id())
+			}
+		} else {
+			q := builder.ChangeComment(c)
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error updating comment for view %v", d.Id())
+			}
+		}
+	}
+	if d.HasChange("is_secure") {
+		secure := d.Get("is_secure")
+
+		if secure.(bool) {
+			q := builder.Secure()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error setting secure for view %v", d.Id())
+			}
+		} else {
+			q := builder.Unsecure()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error unsetting secure for materialized view %v", d.Id())
+			}
+		}
+	}
+
+	if d.HasChange("cluster_by") {
+		cluster_by := d.Get("cluster_by")
+
+		if c := cluster_by.(string); c == "" {
+			q := builder.RemoveClusterBy()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error unsetting cluster_by for view %v", d.Id())
+			}
+		} else {
+			q := builder.ChangeClusterBy(c)
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error updating cluster_by for view %v", d.Id())
+			}
+		}
+	}
+
+	return ReadMaterializedView(d, meta)
+}
+
+// DeleteMaterializedView implements schema.DeleteFunc
+func DeleteMaterializedView(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	dbName, schema, view, err := splitMaterializedViewID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	q := snowflake.MaterializedView(view).WithDB(dbName).WithSchema(schema).Drop()
+
+	err = snowflake.Exec(db, q)
+	if err != nil {
+		return errors.Wrapf(err, "error deleting materialized view %v", d.Id())
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+// MaterializedViewExists implements schema.ExistsFunc
+func MaterializedViewExists(data *schema.ResourceData, meta interface{}) (bool, error) {
+	db := meta.(*sql.DB)
+	dbName, schema, view, err := splitMaterializedViewID(data.Id())
+	if err != nil {
+		return false, err
+	}
+
+	q := snowflake.MaterializedView(view).WithDB(dbName).WithSchema(schema).Show()
+	rows, err := db.Query(q)
+	if err != nil {
+		return false, err
+	}
+	defer rows.Close()
+
+	if rows.Next() {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// splitMaterializedViewID takes the <database_name>|<schema_name>|<view_name> ID and returns the database
+// name, schema name and view name.
+func splitMaterializedViewID(v string) (string, string, string, error) {
```

Comment:
Reading and writing the ids needs to follow the same pattern used elsewhere (using csv). I am currently refactoring it in #380, so after that gets merged we need to refactor this to make use of that.